test-apps/display-test-app/src/backend/Backend.ts

```diff
@@ -251,3 +255,65 @@ export const initializeDtaBackend = async (hostOpts?: ElectronHostOptions & Mobi
   Logger.setLevelDefault(logLevel);
   Logger.setLevel("SVT", LogLevel.Trace);
 };
+
+async function initializeAuthorizationClient(): Promise<AuthorizationClient | undefined> {
+  if (process.env.IMJS_OIDC_HEADLESS) {
+    if (checkEnvVars(
+      "IMJS_OIDC_CLIENT_ID",
+      "IMJS_OIDC_SCOPE",
+      "IMJS_OIDC_EMAIL",
+      "IMJS_OIDC_PASSWORD"
+    )) {
+      // test-only authorization client with credentials from the environment.
+      return new TestBrowserAuthorizationClient({
+        clientId: process.env.IMJS_OIDC_CLIENT_ID!,
+        redirectUri: process.env.IMJS_OIDC_REDIRECT_URI ?? "http://localhost:3000/signin-callback",
+        scope: process.env.IMJS_OIDC_SCOPE!,
+        clientSecret: process.env.IMJS_OIDC_CLIENT_SECRET,
+      }, {
+        email: process.env.IMJS_OIDC_EMAIL!,
+        password: process.env.IMJS_OIDC_PASSWORD!,
+      });
+    } else if (checkEnvVars(
+      "IMJS_OIDC_CLIENT_ID",
+      "IMJS_OIDC_CLIENT_SECRET",
+      "IMJS_OIDC_SCOPE",
+    )) {
+      // TODO: This throws a weird exception at run-time (ServiceAuthorizationClient is unable to load @itwin/core-bentley).
+      return undefined;
+      // // Setup a client using the service authorization workflow.
+      // return new ServiceAuthorizationClient({
+      //   clientId: process.env.IMJS_OIDC_CLIENT_ID!,
+      //   clientSecret: process.env.IMJS_OIDC_CLIENT_SECRET!,
+      //   scope: process.env.IMJS_OIDC_SCOPE!,
+      //   authority: process.env.IMJS_OIDC_AUTHORITY,
+      // });
+    }
+  } else if (checkEnvVars("IMJS_OIDC_CLIENT_ID", "IMJS_OIDC_SCOPE")) {
+    if (ProcessDetector.isElectronAppBackend) {
+      return new ElectronMainAuthorization({
+        clientId: process.env.IMJS_OIDC_CLIENT_ID!,
+        scope: process.env.IMJS_OIDC_SCOPE!,
+        redirectUri: process.env.IMJS_OIDC_REDIRECT_URI ?? "http://localhost:3000/signin-callback",
+      });
+    }
+  }
+  // Note: Mobile's default auth client works, and will be used if we get here on mobile.
+  return undefined;
+}
+
+/**
+ * Logs a warning if only some are provided
+ * @returns true if all are provided, false if any missing.
+ */
+function checkEnvVars(...keys: Array<string>): boolean {
+  const missing = keys.filter((name) => process.env[name] === undefined);
+  if (missing.length === 0) {
```

Comment:
I'd remove these lines and change the return:
```ts
return missing.length === 0;
```

Reply:
Hmm, the highlighted lines didn't work (again). These are the lines I suggested removing:
```ts
  if (missing.length === 0) {
    return true;
  }
```

Reply:
Two things:

1. Your suggestion changes the behavior of the function (in a bad way).
2. This function was copied verbatim from dpta.

Reply:
I guess I misread what you said, and point 1 isn't valid.

Reply:
But due to point 2, I think it should be left alone.

Reply:
Actually, your suggested change does break the function, so I take back saying that I was wrong about that.